### ISSUE TITLE
`RANSAC(score_type="msac")` finds no model at `inl_th=2` with 50% outliers: the MSAC score is used as an inlier count

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4868]
With `score_type="msac"`, `RANSAC.verify` scores a model as `N − Σ clamp(err, 0, inl_th²)`, a count minus squared pixels. `forward` then uses that number as an inlier count in two places. It accepts a model only if the score exceeds `best_score_total`, which starts at `minimal_sample_size` (4 for a homography). It also passes `int(best_score_total)` to `max_samples_by_conf` as `n_inl` for early stopping. Every outlier contributes `−inl_th²`, so with `n_out` outliers even the true model scores at most `N − n_out · inl_th²`. On 16 exact matches plus 16 random outliers, the true homography scores −32 at the default `inl_th = 2.0` and −364 at `inl_th = 5`. `forward` returns the all-zero model with an empty mask, while `score_type="ransac"` finds the model at every threshold. MSAC (Torr & Zisserman, 2000) minimises `Σ min(e², T²)`. That value ranks models against each other, but it is not a number of inliers and cannot be compared with a sample size.

## Repro

```python
import torch
from kornia.geometry.linalg import transform_points
from kornia.geometry.ransac import RANSAC

H = torch.tensor([[[1.034685, -0.04160598, -63.61482],
                   [0.1039753, 1.135451, -62.87412],
                   [2.851519e-04, 1.404178e-04, 1.0]]])
gx, gy = torch.tensor([160.0, 230.0, 330.0, 440.0]), torch.tensor([95.0, 150.0, 240.0, 320.0])
p1 = torch.stack(torch.meshgrid(gx, gy, indexing="xy"), -1).reshape(16, 2) + torch.linspace(0, 3, 16)[:, None]
p2 = transform_points(H, p1[None])[0]                       # 16 exact matches
g = torch.Generator().manual_seed(1)
o1, o2 = torch.rand(16, 2, generator=g) * 400 + 50, torch.rand(16, 2, generator=g) * 400 + 50   # 16 outliers
kp1, kp2 = torch.cat([p1, o1]), torch.cat([p2, o2])

for th in (1.0, 2.0, 5.0):
    for score in ("ransac", "msac"):
        model, inl = RANSAC("homography", inl_th=th, score_type=score, seed=0)(kp1, kp2)
        print(f"inl_th={th} score_type={score:6}: model found = {bool(model.abs().sum() > 0)!s:5}, inliers = {int(inl.sum())}/32")

for th in (1.0, 2.0, 5.0):
    r = RANSAC("homography", inl_th=th, score_type="msac")
    _, _, score, n_inl = r.verify(kp1, kp2, H, r.inl_th**2)   # the true H, scored as forward scores it
    print(f"true H, inl_th={th}: msac score = {score:.1f}, inlier count = {n_inl}, bar the score must exceed = {r.minimal_sample_size}")
# all-inlier control: the defect needs outliers (each outlier contributes -inl_th**2 to the score)
model, inl = RANSAC("homography", inl_th=5.0, score_type="msac", seed=0)(p1, p2)
print(f"16 matches, no outliers, inl_th=5, msac: model found = {bool(model.abs().sum() > 0)}, inliers = {int(inl.sum())}/16")
```

```text
inl_th=1.0 score_type=ransac: model found = True , inliers = 16/32
inl_th=1.0 score_type=msac  : model found = True , inliers = 16/32
inl_th=2.0 score_type=ransac: model found = True , inliers = 16/32
inl_th=2.0 score_type=msac  : model found = False, inliers = 0/32
inl_th=5.0 score_type=ransac: model found = True , inliers = 17/32
inl_th=5.0 score_type=msac  : model found = False, inliers = 0/32
true H, inl_th=1.0: msac score = 16.0, inlier count = 16, bar the score must exceed = 4
true H, inl_th=2.0: msac score = -32.0, inlier count = 16, bar the score must exceed = 4
true H, inl_th=5.0: msac score = -364.1, inlier count = 17, bar the score must exceed = 4
16 matches, no outliers, inl_th=5, msac: model found = True, inliers = 16/16
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same output), CPU float32, macOS arm64.

The defect needs outliers. Without them the score stays close to N and the model is found (last line), so an all-inlier test cannot detect it. MSAC can succeed on this data only while `32 − 16 · inl_th² > 4`, i.e. `inl_th < 1.32`; the same call finds the model at `inl_th = 1.30` and returns none at `1.35`. The MSAC scorer was added in #3458.

## Classification

Clearly broken output: a documented `score_type` returns no model at the default threshold on data that `score_type="ransac"` solves.

Idiom 3 (a guard whose predicate does not match the documented contract): the acceptance test `model_score > best_score_total` and the early-stopping bound treat the MSAC score as an inlier count.

## Disposition

Focused fix, welcome as a PR: keep the MSAC score only for ranking models against each other. Initialise the running best score to `-inf` rather than to `minimal_sample_size`; the existing `num_inliers >= polisher_sample_size` condition already gates acceptance on a count. Pass the best model's `num_inliers`, not its score, to `max_samples_by_conf`. The `ransac` score path is unchanged, because there the score is the count.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Related: #4871 (the (N, 1).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
